spec/support/platforms.rb

```diff
@@ -79,7 +79,7 @@ def local_engine_version
       when "jruby"
         JRUBY_VERSION
       else
-        raise BundlerError, "That RUBY_ENGINE is not recognized"
```

Comment:
Woo! This is awesome! 🎉 

My first question though why are you modifying this behavior instead of adding to it?

Reply:
we might also want to check that `RUBY_ENGINE_VERSION` is indeed defined? I'm not sure if all Rubies define it

Reply:
My thinking was rather than adding a special case for `truffleruby` I'd make it general.
Every not ancient (=1.8) Ruby version should define `RUBY_ENGINE_VERSION`, it's a standard constant like `RUBY_ENGINE`, `RUBY_VERSION`, etc.
I can add a defined? it it causes a problem for an implementation but it seems unlikely as currently this code path was hardcoded to only support MRI, JRuby and Rubinius.

Reply:
@colby-swandale What do you think? Is this code fine as-is or should I change it something else?

Reply:
I'm happy with the PR, so let's merge it